src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private final double firstLeg;
+    private final double secondLeg;
+    private final ColorSupplier colorsupplier = new ColorSupplier();
```

Comment:
Don't use verbs for class/interface names. In this case, 'ColorSupplier' should be a noun, something like 'ColorProvider' or 'ColorGenerator'.